### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by LukeLoze1**
## Motivation and Context
unit testing for rabbit_helper and subscriber

## What has changed
Unit tests added for them

## How to test?
pipenv install
pipenv shell
pytest test/

## Links
https://trello.com/b/E9lArxTH/ras-rm-orange

### CODE REVIEW

File: test/test_rabbit_helper.py [hunk 1]
```diff
@@ -1,43 +1,92 @@
...
+                          binding_key=RM_BINDING_KEY,
+                          exchange_name=RM_RABBIT_EXCHANGE,
+                          queue_name=RM_RABBIT_QUEUE,
+                          queue_args=self.queue_args)
+
+            channel_mock.exchange_declare.assert_called_once_with(exchange=RM_RABBIT_EXCHANGE, exchange_type='direct',
+                                                                  durable=True)
+            channel_mock.queue_declare.assert_called_once_with(arguments=self.queue_args, durable=True,
+                                                               queue=RM_RABBIT_QUEUE)
+            channel_mock.queue_bind(exchange=RM_RABBIT_EXCHANGE, queue=RM_RABBIT_QUEUE, routing_key=RM_BINDING_KEY)
```
Comment (rjweeks) [inline test/test_rabbit_helper.py]: Needs .assert_called_once_with

**Review by rjweeks — CHANGES_REQUESTED** [review]

File: test/test_rabbit_helper.py [hunk 2]
```diff
@@ -1,43 +1,92 @@
...
+
+    def setUp(self):
+        test_environment_variables = {
+            'RABBIT_AMQP': RM_RABBIT_AMQP,
+            'RABBIT_ROUTING_KEY': RM_BINDING_KEY,
+            'RABBIT_QUEUE': RM_RABBIT_QUEUE,
+            'RABBIT_EXCHANGE': RM_RABBIT_EXCHANGE,
+        }
+        os.environ.update(test_environment_variables)
+
```
Comment (rjweeks) [inline test/test_rabbit_helper.py:33]: Fix PEP8 warnings.

**Review by rjweeks — DISMISSED** [review]

File: test/helpers/when_then_return.py [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
 
     return when_then_return
+
+
+def create_when_then_return_kwargs(expected_kwargs, return_value):
+    def when_then_return(*args, **kwargs):
+        assert expected_kwargs == kwargs
+        return return_value
+
+    return when_then_return
```
Comment (AdamHawtin) [inline test/helpers/when_then_return.py]: These two functions can be combined if you make `expected_kwargs` a kwarg itself with default of an empty dictionary and check both the expected args and kwargs. Also something like `create_stub_function` might be a clearer name, it took me a little while to realise exactly what the purpose of these was.

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
Tests seem good, though to run in PyCharm you do have to make sure the working directory is set to the project directory or the `subscriber` can't find its message templates.

Also, the README could use some direction on running the unit tests. Perhaps a Makefile would be useful like we've used in the other python projects just to collect together the various shell commands into simple targets like `make test`.

File: test/helpers/when_then_return.py (hunk 3 above)
Comment (jcox-dev) [reply test/helpers/when_then_return.py]: @AdamHawtin Solid suggestions. I've refactored and updated the README in the [latest commit](https://github.com/ONSdigital/census-rm-pubsub/pull/8/commits/e7381f7153d16f155077c432dac5d393ecda6b72).

**Review by AdamHawtin — APPROVED** [review]

**Pull request merged**